src/librustc_errors/lib.rs

```diff
@@ -251,21 +252,24 @@ impl Handler {
     pub fn with_tty_emitter(color_config: ColorConfig,
                             can_emit_warnings: bool,
                             treat_err_as_bug: bool,
+                            macro_backtrace: bool,
                             cm: Option<Rc<CodeMapper>>)
                             -> Handler {
         let emitter = Box::new(EmitterWriter::stderr(color_config, cm, false));
-        Handler::with_emitter(can_emit_warnings, treat_err_as_bug, emitter)
+        Handler::with_emitter(can_emit_warnings, treat_err_as_bug, macro_backtrace, emitter)
     }
 
     pub fn with_emitter(can_emit_warnings: bool,
```

Comment:
Could you add another method rather than adding another bool flag to this one? That would be more self-documenting (bools tell the reader nothing), less susceptible to bugs (since it is easy enough to confuse the two bools here), and avoids a breaking change.

Reply:
Do you mean a new method like `Handler::with_backtracing_emitter(...)`? We also thought of always initializing the flag to `false` and providing a setter. But that's kind of a weird mixture of constructor flags and setters which seems like it ought to be refactored into a settings struct anyway.

Reply:
Maybe `with_emitter_with_settings` and take a settings struct to allow adding more flags in the future which are similarly detailed.

Reply:
I'm calling it `with_emitter_and_flags` and this should reduce the size of the patch / avoid breaking rustfmt. But we should still refactor users of `with_emitter`/`with_tty_emitter` and remove those at some point!